storage/backup/backup-cli/src/backup_types/transaction/restore.rs

```diff
@@ -393,14 +393,15 @@ impl TransactionRestoreBatchController {
                         let txns_to_save: Vec<_> = txns.drain(..num_to_save).collect();
                         let txn_infos_to_save: Vec<_> = txn_infos.drain(..num_to_save).collect();
                         let event_vecs_to_save: Vec<_> = event_vecs.drain(..num_to_save).collect();
-                        write_sets.drain(..num_to_save);
+                        let ws_to_save = write_sets.drain(..num_to_save).collect();
```

Comment:
nit: calling it `write_sets_to_save` seems slightly better because it's symmetric to the other lines.